api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,448 @@
+	// most one IP address per IP stack used). The order of stacks should be the same as order
+	// of subnets in Cluster Networks, Service Networks, and Machine Networks.
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	IngressVIPs []string `json:"ingressVIPs,omitempty"`
+
+	// HoldInstallation will prevent installation from happening when true.
+	// Inspection and validation will proceed as usual, but once the RequirementsMet condition is true,
+	// installation will not begin until this field is set to false.
+	// +kubebuilder:default:=false
+	// +optional
+	HoldInstallation bool `json:"holdInstallation,omitempty"`
+
+	// AdditionalNTPSources is a list of NTP sources (hostname or IP) to be added to all cluster
+	// hosts. They are added to any NTP sources that were configured through other means.
+	// +optional
+	AdditionalNTPSources []string `json:"additionalNTPSources,omitempty"`
+
+	// MachineNetwork is the list of IP address pools for machines.
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+
+	// ClusterNetwork is the list of IP address pools for pods.
+	// +optional
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork,omitempty"`
+
+	// ServiceNetwork is the list of IP address pools for services.
+	// +optional
+	ServiceNetwork []string `json:"serviceNetwork,omitempty"`
+
+	// NetworkType is the Container Network Interface (CNI) plug-in to install
+	// The default value is OpenShiftSDN for IPv4, and OVNKubernetes for IPv6 or SNO
+	// +kubebuilder:validation:Enum=OpenShiftSDN;OVNKubernetes
+	// +kubebuilder:default:=OVNKubernetes
+	// +optional
+	NetworkType string `json:"networkType,omitempty"`
+
+	// Additional cluster-wide annotations to be applied to the rendered templates
+	// +optional
+	CrAnnotations CrAnnotations `json:"crAnnotations,omitempty"`
+
+	// ClusterLabels is used to assign labels to the cluster to assist with policy binding.
+	// +optional
+	ClusterLabels map[string]string `json:"clusterLabels,omitempty"`
+
+	// InstallConfigOverrides is a generic way of passing install-config parameters.
+	// +optional
+	InstallConfigOverrides string `json:"installConfigOverrides,omitempty"`
+
+	// Json formatted string containing the user overrides for the initial ignition config
+	// +optional
+	IgnitionConfigOverride string `json:"ignitionConfigOverride,omitempty"`
+
+	// DiskEncryption is the configuration to enable/disable disk encryption for cluster nodes.
+	// +optional
+	DiskEncryption DiskEncryption `json:"diskEncryption,omitempty"`
+
+	// ProxySettings defines the proxy settings used for the install config
+	// +optional
+	ProxySettings aiv1beta1.Proxy `json:"proxy,omitempty"`
+
+	// ExtraManifestsRefs is list of config map references containing additional manifests to be applied to the cluster.
+	// +optional
+	ExtraManifestsRefs []corev1.LocalObjectReference `json:"extraManifestsRef,omitempty"`
```

Comment:
I see it as replacing functionalities of [ExtraManifests](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfig.go#L199), where we could list the git paths to look for extra-manifests and the capability to filter them e.g. [include/exclude](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/gitops-subscriptions/argocd/ExtraManifestsFilter.md), is there any doc explaining how the current ztp behavior would be translated here?

Reply:
The original `SiteConfig` CR supports filtering of and search paths for the extra manifests through `spec.clusters[].extraManifests`. These features were meaningful in the context of GitOps where the extra manifests ConfigMap was built from the “bare” CRs in the git repository. These CRs can’t exist on the hub cluster as “bare” CRs, so users must ensure they are built into a ConfigMap on the hub cluster. We drop support for extraManifest filtering since the user can create any number of extra manifest ConfigMaps with the proper content per OCP release, or based on variants of their deployment. Each `SiteConfig` can opt into the desired ConfigMap.